packages/compass-components/src/hooks/use-toast.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { createContext, useCallback, useContext, useState } from 'react';
+import type { ToastVariant } from '..';
+import { css } from '..';
+import { Toast } from '..';
+
+type ToastProperties = {
+  title?: React.ReactNode;
+  body: React.ReactNode;
+  variant: ToastVariant;
+  progress?: number;
+  timeout?: number;
+};
+
+type ToastState = ToastProperties & {
+  timeoutRef?: ReturnType<typeof setTimeout>;
+};
+
+type ToastId = string;
+
+type ToastContextState = {
+  toasts: Record<ToastId, ToastState>;
+  setToasts: (toasts: Record<ToastId, ToastState>) => void;
+};
+
+interface ToastActions {
+  openToast: (id: ToastId, toastProperties: ToastProperties) => void;
+  closeToast: (id: ToastId) => void;
+}
+
+const ToastContext = createContext<ToastContextState>({
+  toasts: {},
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  setToasts: (t: Record<ToastId, ToastProperties>) => {
+    return;
+  },
+});
+
+const toastStyles = css({
+  button: {
+    position: 'absolute',
+  },
+});
+
+const ToastPile = (): React.ReactElement => {
+  const { toasts } = useContext(ToastContext);
+  const { closeToast } = useGlobalToast();
+
+  return (
+    <>
+      {Object.entries(toasts).map(
+        ([id, { title, body, variant, progress }]) => (
+          <Toast
+            className={toastStyles}
+            key={id}
+            title={title}
+            body={body}
+            variant={variant}
+            progress={progress}
+            open={true}
+            close={() => closeToast(id)}
+          />
+        )
+      )}
+    </>
+  );
+};
+
+/**
+ * @example
+ *
+ * ```
+ * const MyButton = () => {
+ *   const { openToast } = useToast('my-namespace');
+ *   return <button onClick={() => openToast(
+ *      'myToast1', {title: 'This is a notification'})} />
+ * };
+ *
+ * <ToastArea><MyButton/><ToastArea>
+ * ```
+ *
+ * @returns
+ */
+export const ToastArea = ({
+  children,
+}: {
+  children?: React.ReactNode;
+}): React.ReactElement => {
+  const [toasts, setToasts] = useState({});
+
+  return (
+    <ToastContext.Provider value={{ toasts, setToasts }}>
+      <>{children}</>
+      <ToastPile />
+    </ToastContext.Provider>
+  );
+};
+
+function useGlobalToast(): ToastActions {
+  const { toasts, setToasts } = useContext(ToastContext);
```

Comment:
It's generally fine as it is right now, but as both methods exported from `useGlobalToast` need to be defined only once, they would be better candidates to get passed to the provider, instead of just `useState` return values. I would probably go about it like this. The `useGlobalToast` hook defines the toast state instead of getting it from context ...

```suggestion
function useGlobalToast(): ToastActions {
  const [toasts, setToasts] = useState({});
```

... and is used inside the `ToastArea` component and then open/close methods are the things that are passed in the context and accessed in the `useToast` hook

Reply:
Nice, will change it